pkg/interactors/register_thing.go

```diff
@@ -0,0 +1,124 @@
+package interactors
+
+import (
+	"strconv"
+
+	"github.com/CESARBR/knot-babeltower/pkg/logging"
+	"github.com/CESARBR/knot-babeltower/pkg/network"
+)
+
+// RegisterThing use case to register a new thing
+type RegisterThing struct {
+	logger       logging.Logger
+	msgPublisher network.Publisher
+}
+
+// ErrorIDLenght is raised when ID is more than 16 characters
+type ErrorIDLenght struct{}
+
+// ErrorIDInvalid is raised when ID is not in hexadecimal value
+type ErrorIDInvalid struct{}
+
+// ErrorNameNotFound is raised when Name is empty
+type ErrorNameNotFound struct{}
+
+// ErrorArgument is raised when Name is empty
+type ErrorArgument struct{ msg string }
+
+func (err ErrorIDLenght) Error() string {
+	return "ID length error"
+}
+
+func (err ErrorIDInvalid) Error() string {
+	return "ID is not in hexadecimal"
+}
+
+func (err ErrorNameNotFound) Error() string {
+	return "Name not found"
+}
+
+func (err ErrorArgument) Error() string {
+	return err.msg
+}
+
+// NewRegisterThing contructs the use case
+func NewRegisterThing(logger logging.Logger, msgPublisher network.Publisher) *RegisterThing {
+	return &RegisterThing{logger, msgPublisher}
+}
+
+func (rt *RegisterThing) reply(id, token string, err error) error {
+	var errStr *string
```

Comment:
(asking because I don't really know) 
What is the reason to use a string pointer in this case? Would it be to pass a nil value when necessary?

Reply:
The behavior JSON response is that property `error` is `null` or a string.
When using the golang type `error` if the value is `nil` the JSON parsed will be `{}` and not `null` since the error is actually an `interface{}`.
See this [link](https://stackoverflow.com/questions/31048557/assigning-null-to-json-fields-instead-of-empty-strings)